### PR TITLE
[#572] Allow image url with more than 2 slashes to be parsed properly.

### PR DESCRIPTION
**Pull request by mnovak1**
Please make sure your PR meets the following requirements:
- [x] Pull Request contains a description of the changes
- [x] Pull Request does not include fixes for multiple issues/topics
- [x] Code is formatted, imports ordered, code compiles and tests are passing
- [x] Code is self-descriptive and/or documented

CI runs:  
Without fix: https://url.corp.redhat.com/a7db151
With fix: https://url.corp.redhat.com/bbed950
* GalleonDatasourceNoDriverVersionTest#noMsSQLDriverVersion,GalleonDatasourceFeaturePackMsSQLTest - caused by not updated mssql image
* Rerun HAServletCounterUpstreamKubePingTest,RollingUpdatesKubePingTest → https://url.corp.redhat.com/e8c1138 passed

### CODE REVIEW

File: core/src/test/java/cz/xtf/core/image/ImageTest.java [hunk 1]
```diff
@@ -99,4 +99,50 @@ public void testNotResolvableImage() {
...
+        Assertions.assertEquals("nginx", image.getRepo(), "Wrong repository name.");
+        Assertions.assertTrue(image.getTag().isEmpty(), "Tag for image url: 'nginx' must be empty.");
+    }
+
+    @Test
+    public void testImageFromUrlTwoTokens() {
+        Image image = Image.from("user/nginx");
+        Assertions.assertTrue(image.getRegistry().isEmpty(), "Registry for image url: 'user/nginx' must be empty.");
+        Assertions.assertEquals(image.getUser(), "user", "User for image url: 'user/nginx' is wrong.");
+        Assertions.assertEquals("nginx", image.getRepo(), "Wrong repository name.");
```
Comment (simkam) [inline core/src/test/java/cz/xtf/core/image/ImageTest.java:117]: weird order of expected/actual

> [assertEquals](https://junit.org/junit5/docs/5.0.1/api/org/junit/jupiter/api/Assertions.html#assertEquals-byte-byte-java.lang.String-)(byte expected, byte actual, [String](https://docs.oracle.com/javase/8/docs/api/java/lang/String.html?is-external=true) message)

below occurrences as well

**Review by simkam — CHANGES_REQUESTED** [review]

File: core/src/test/java/cz/xtf/core/image/ImageTest.java (hunk 1 above)
Comment (mnovak1) [reply core/src/test/java/cz/xtf/core/image/ImageTest.java]: Thanks, fixed.

**Review by simkam — APPROVED** [review]

File: core/src/main/java/cz/xtf/core/image/Image.java [hunk 2]
```diff
@@ -70,7 +71,10 @@ public static Image from(String imageUrl) {
                 repoTag = slashTokens[2];
                 break;
             default:
-                throw new IllegalArgumentException("image '" + imageUrl + "' should have one or two '/' characters");
+                registry = slashTokens[0];
+                user = slashTokens[1];
```
Comment (mchoma) [inline core/src/main/java/cz/xtf/core/image/Image.java:75]: This is different to what dosu says :) My understanding is aligned with dosu implementation.

IIUC dosu implementation https://github.com/xtf-cz/xtf/issues/572#issuecomment-2296478769 will bring this result
```
registry = mcr.microsoft.com
user = mssql/rhel
repoTag = server:2022-CU13-rhel-9.1
```

whereas your implementation will bring 

```
registry = mcr.microsoft.com
user = mssql
repoTag = rhel/server:2022-CU13-rhel-9.1
```

Comment (simkam) [reply core/src/main/java/cz/xtf/core/image/Image.java:75]: to me the latter makes more sense, though we are only bending image url (which can be almost anything) to what the class expects

Comment (mchoma) [reply core/src/main/java/cz/xtf/core/image/Image.java:75]: Yes, you was faster then me wanted to write `user = mssql ` makes more sense to me now while thinking about it.

**Review by mchoma — APPROVED** [review]
I consider https://github.com/xtf-cz/xtf/issues/572 for addressed. Now it is enabled to use image url with more than 2 slashes.

Tests proove results are expected.

**Pull request merged**